util/deephash/deephash.go

```diff
@@ -348,92 +334,91 @@ func genTypeHasher(ti *typeInfo) typeHasherFunc {
 	case reflect.Struct:
 		return genHashStructFields(t)
 	case reflect.Map:
-		return func(h *hasher, v addressableValue) {
+		return func(h *hasher, p pointer) {
+			v := p.asValue(t).Elem() // reflect.Map kind
 			if v.IsNil() {
 				h.HashUint8(0) // indicates nil
 				return
 			}
 			if ti.isRecursive {
-				ptr := pointerOf(v)
-				if idx, ok := h.visitStack.seen(ptr); ok {
+				pm := v.UnsafePointer() // underlying pointer of map
+				if idx, ok := h.visitStack.seen(pm); ok {
 					h.HashUint8(2) // indicates cycle
 					h.HashUint64(uint64(idx))
 					return
 				}
-				h.visitStack.push(ptr)
-				defer h.visitStack.pop(ptr)
+				h.visitStack.push(pm)
+				defer h.visitStack.pop(pm)
 			}
 			h.HashUint8(1) // indicates visiting a map
-			h.hashMap(v, ti, ti.isRecursive)
+			h.hashMap(v, ti)
 		}
 	case reflect.Pointer:
 		et := t.Elem()
-		if typeIsMemHashable(et) {
```

Comment:
Unnecessary since we check for mem-hashable types at the start of `genTypeHasher`.